contrib/utilities/checkdoxygen.py

```diff
@@ -59,6 +59,12 @@ def check_multiple_defined_headers(lines):
             thisheader = l                             # take this line
             thisheader = thisheader.replace(' ', '')   # remove all whitespace
             thisheader = thisheader.split('<h')[1]     # remove header tag '<h'
+
+            if thisheader[0] == '2':
+                # We disable h2 headers in tutorials: they don't show up in
+                # the table of contents.
```

Comment:
The reason they don't show up is because we/the script in question doesn't put them into the ToC. And the reason it doesn't is fundamentally because we have to make a decision on how to indent the ToC into sections and subsections. We just made the decision that H1 is the program, H2 is Introduction/Commented Program/Results/Plain Progam, and H3...H5 is whatever the author wants to do.

Maybe you can put that into the comment here in some form or other?

Reply:
> H2 is Introduction/Commented Program/Results/Plain Progam

That is not correct, we use h1 for that. So I am not sure what the motivation for avoiding h2 is.